vars/pythonSetupPyPipeline.groovy

```diff
@@ -24,23 +24,35 @@ def call(Map pipelineParams) {
     pipelineParams["dockerFilename"] = "Dockerfile"
   }
   if (!pipelineParams.dockerBuildArgs) {
-    pipelineParams["dockerBuildArgs"] = ""
+    pipelineParams["dockerBuildArgs"] = "--no-cache --network host"
   }
   if (!pipelineParams.dockerRunArgs) {
     pipelineParams["dockerRunArgs"] = "-v /etc/passwd:/etc/passwd:ro -v /opt/jenkins/.ssh:/opt/jenkins/.ssh:ro --network host"
   }
 
-  pipeline {
-    agent {
-      dockerfile {
-        filename pipelineParams.dockerFilename
-        additionalBuildArgs pipelineParams.dockerBuildArgs
-        args pipelineParams.dockerRunArgs
-      }
+  if (pipelineParams.dockerDeploy) {
+    if (!pipelineParams.dockerRegistryUrl) {
+      error("${LOG_TAG} Please provide a Docker registry URL (eg. https://registry.example.com)")
+    }
+    if (!pipelineParams.dockerRegistryCredentialsId) {
+      error("${LOG_TAG} Please provide a Jenkins credentials id for deploying to ${pipelineParams.dockerRegistryUrl}")
```

Comment:
I don't think we should be tying credential usage in Jenkins to the Jenkins credential store.
This effectively precludes the use of Hashicorp Vault (or other sources).

Reply:
I won't fix this now as we already use pypiCredentials in the same way. Will create a ticket for those two items

Reply:
I'm fine with that, please link the ticket here :+1:

Reply:
See #14